src/stop_motion_obj.py

```diff
@@ -729,58 +746,83 @@ def getMeshIdxFromFrameNumber(_obj, frameNum):
+        swapMeshAndMaterials(_obj, next_mesh, prev_mesh_materials, mss.showAsSingleMesh)
 
-def setFrameObjStreamed(obj, frameNum, forceLoad=False, deleteMaterials=False):
+
+def setFrameObjStreamed(obj, frameNum, updateSingleMesh, forceLoad=False, deleteMaterials=False):
+    
     mss = obj.mesh_sequence_settings
-    idx = getMeshIdxFromFrameNumber(obj, frameNum)
-    nextMeshProp = getMeshPropFromIndex(obj, idx)
 
-    # if we want to load new meshes as needed and it's not already loaded
-    if nextMeshProp.inMemory is False and (mss.streamDuringPlayback is True or forceLoad is True):
-        importStreamedFile(obj, idx)
-        if deleteMaterials is True:
+    if(not(updateSingleMesh is False and mss.showAsSingleMesh is True)):        
+        idx = getMeshIdxFromFrameNumber(obj, frameNum)
+        nextMeshProp = getMeshPropFromIndex(obj, idx)
+
+        # if we want to load new meshes as needed and it's not already loaded
+        if nextMeshProp.inMemory is False and (mss.streamDuringPlayback is True or forceLoad is True):
+            importStreamedFile(obj, idx)
+            if deleteMaterials is True:
+                next_mesh = getMeshFromIndex(obj, idx)
+                deleteLinkedMeshMaterials(next_mesh)
+
+        # if the mesh is in memory, show it
+        if nextMeshProp.inMemory is True:
             next_mesh = getMeshFromIndex(obj, idx)
-            deleteLinkedMeshMaterials(next_mesh)
-
-    # if the mesh is in memory, show it
-    if nextMeshProp.inMemory is True:
-        next_mesh = getMeshFromIndex(obj, idx)
-
-        # store the current mesh for grabbing the material later
-        prev_mesh = obj.data
-        if next_mesh != prev_mesh:
-            # swap the old one with the new one
-            obj.data = next_mesh
-
-            # if we need to, copy the materials from the old one onto the new one
-            if obj.mesh_sequence_settings.perFrameMaterial is False:
-                if len(prev_mesh.materials) > 0:
-                    obj.data.materials.clear()
-                    for material in prev_mesh.materials:
-                        obj.data.materials.append(material)
-
-    if mss.cacheSize > 0 and mss.numMeshesInMemory > mss.cacheSize:
-        idxToDelete = nextCachedMeshToDelete(obj, idx)
-        if idxToDelete >= 0:
-            removeMeshFromCache(obj, idxToDelete)
 
+        # store the current materials for grabbing them later
+        prev_mesh_materials = []
+        for material in obj.data.materials:
+            prev_mesh_materials.append(material.copy())
+
+        #Set meshes and materials
+        swapMeshAndMaterials(obj, next_mesh, prev_mesh_materials, mss.showAsSingleMesh)
+
+        if mss.cacheSize > 0 and mss.numMeshesInMemory > mss.cacheSize:
+            idxToDelete = nextCachedMeshToDelete(obj, idx)
+            if idxToDelete >= 0:
+                removeMeshFromCache(obj, idxToDelete)
+
+def swapMeshAndMaterials(old_Object, new_Mesh, old_mesh_materials, forSingleMesh):
```

Comment:
Minor change: `oldObject` and `newMesh`

I know that my code is pretty inconsistent in that regard, but I've had 3 different jobs with 3 different code styles during the development of this addon. I'm trying to slowly move toward this style when possible.